### PR TITLE
fix getParcelable deprecation

### PR DESCRIPTION
**Pull request by RobozinhoD**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
part of https://github.com/ankidroid/Anki-Android/issues/12364

## Approach
_How does this change address the problem?_

## How Has This Been Tested?

I don't know how to test this

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt [hunk 1]
```diff
@@ -142,5 +142,9 @@ class CompatHelper private constructor() {
         fun PackageManager.resolveServiceCompat(intent: Intent, flags: ResolveInfoFlagsCompat): ResolveInfo? {
             return compat.resolveService(this, intent, flags)
         }
+
+        inline fun <reified T> Bundle.getParcelableCompat(key: String): T? {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt:146]: Would be great to have comments here

**Review by david-allison — CHANGES_REQUESTED** [review]
nitpick, then let's go!

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt (hunk 1 above)
Comment (RobozinhoD) [reply AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt:159]: done

**Comment by mikehardy** [comment]
Conflicts with a previous change that you also made, in parallel I think ? (no good deed unpunished! hopefully easy to resolve)

While de-conflicting, please rebase against main just to make sure ktlint changes from #13265 are handled just in case

Thank you!

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatV21.kt [hunk 2]
```diff
@@ -91,6 +91,10 @@ open class CompatV21 : Compat {
         return packageManager.resolveService(intent, flags.value.toInt())
     }
 
+    override fun <T> getParcelable(bundle: Bundle, key: String?, clazz: Class<T>): T? {
+        return bundle.getParcelable(key)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/compat/CompatV21.kt:95]: Can this be tested? I'm surprised it works without a cast

**Comment by david-allison** [comment]
Lint is failing

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatV21.kt (hunk 2 above)
Comment (RobozinhoD) [reply AnkiDroid/src/main/java/com/ichi2/compat/CompatV21.kt:95]: how can I test this?

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/compat/CompatV21.kt:95]: On an emulator using an API before the method in `CompatV33` overrides it

Comment (RobozinhoD) [reply AnkiDroid/src/main/java/com/ichi2/compat/CompatV21.kt:95]: I added a getParcelableCompat line among DeckPicker and it didn't crash after running the line

**Review by david-allison — APPROVED** [review]
Ah, underlying method is `<T>`, that works

LGTM

**Review by BrayanDSO — APPROVED** [review]
Seems like you really don't like deprecations. LGTM

**Comment by mikehardy** [comment]
I hate deprecations as they *do* eventually result in "cannot release anymore" blockers, so I love these PRs. Thank you!

**Pull request merged**